lib/spack/spack/fetch_strategy.py

```diff
@@ -940,6 +940,126 @@ def __str__(self):
         return '[git] {0}'.format(self._repo_info())
 
 
+@fetcher
+class CvsFetchStrategy(VCSFetchStrategy):
+
+    """Fetch strategy that gets source code from a CVS repository.
+       Use like this in a package:
+
+           version('name',
+                   cvs=':pserver:anonymous@www.example.com:/cvsroot%module=modulename')
+
+       Optionally, you can provide a branch and/or a date for the URL:
+
+           version('name',
+                   cvs=':pserver:anonymous@www.example.com:/cvsroot%module=modulename',
+                   branch='branchname', date='date')
+
+    Repositories are checked out into the standard stage source path directory.
+    """
+    url_attr = 'cvs'
+    optional_attrs = ['branch', 'date']
+
+    def __init__(self, **kwargs):
+        # Discards the keywords in kwargs that may conflict with the next call
+        # to __init__
+        forwarded_args = copy.copy(kwargs)
+        forwarded_args.pop('name', None)
+        super(CvsFetchStrategy, self).__init__(**forwarded_args)
+
+        self._cvs = None
+        self._expect = None
+        if self.branch is not None:
+            self.branch = str(self.branch)
+        if self.date is not None:
+            self.date = str(self.date)
+
+    @property
+    def cvs(self):
+        if not self._cvs:
+            self._cvs = which('cvs', required=True)
+        return self._cvs
+
+    @property
+    def expect(self):
+        if not self._expect:
+            self._expect = which('expect', required=True)
```

Comment:
Is `expect` the only way to run these commands? It seems to be pretty common on Linux/macOS so it might be fine, but if there were a way to avoid another external dependency that would be great. Speaking of which, you should add `cvs` (and possibly `expect`) to the Prerequisites listed in the Getting Started docs.

Reply:
`expect` is necessary to enter a dummy password when connecting to a remote server. There doesn't seem to be a way to specify the password on the command line or on stdin.

Reply:
Well, that's annoying, but if anyone needs to install CVS anyway they can always install expect if they need to.